### DISCUSSION TITLE
Making model_fields optional when doing an update

### DISCUSSION THREAD

**Discussion by Lodimup** [vitalik/django-ninja#379] [Q&A] [answered]
```
class UserInUpdate(ModelSchema):
    class Config:
        model = User
        model_fields = ['first_name', 'last_name', 'email' ]
```

```
@api.put('/users/{username}')
def update_user(request, username: str, payload: UserInUpdate):
    user = get_object_or_404(User, username=username)
    for attr, value in payload.dict().items():
        setattr(user, attr, value)
    user.save()
    
    return {"success": True}
```

Let's say I only want to update email. How do I make it so I don't have to post other fields? 
Thanks!

**Comment by stvdrsch** [comment]
Somebody that found a solution on how to do this?

**Reply by sebastian-philipp** [reply]
https://fastapi.tiangolo.com/tutorial/body-updates/#partial-updates-with-patch ?

**Reply by Lodimup** [reply, marked as answer]
https://django-ninja.rest-framework.com/guides/response/django-pydantic/

here, under Making fields optional

It's implemented on May 28
@[vitalik](https://github.com/vitalik) vitalik
[ v0.22.0](https://github.com/vitalik/django-ninja/tree/v0.22.0)
 https://github.com/vitalik/django-ninja/commit/9c6f13654a3a099e7bd119f8d41d126ae2367d8a

[0.22.0](https://github.com/vitalik/django-ninja/releases/tag/v0.22.0)

**Comment by noamsto** [comment]
If you want to make it work with `Schema` based models, I changed a little bit @uriyyo solution from here https://github.com/pydantic/pydantic/discussions/3089#discussioncomment-8052305

```python
from typing import Any
from ninja import Schema


class User(Schema):
    first_name: str
    last_name: str
    age: int
    address: str

class OptionalSchema(Schema):
    @classmethod
    def __pydantic_init_subclass__(cls, **kwargs: Any) -> None:
        super().__pydantic_init_subclass__(**kwargs)

        for field in cls.model_fields.values():
            field.default = None

        cls.model_rebuild(force=True)
        

class PartialUser(User, OptionalSchema):
    ...
```
